stem_cell/differentiator.py

```diff
@@ -1,18 +1,30 @@
 class Differentiator:
     PAYLOAD_ENTRY_ANSIBLE_PLAYBOOK = '/main.yml'
 
-    def __init__(self, user_data_gateway, version_control_system, ansible):
+    def __init__(self, user_data_gateway, version_control_system, ansible, file_gateway):
         self.ansible = ansible
+        self.file_gateway = file_gateway
         self.user_data_gateway = user_data_gateway
         self.version_control_system = version_control_system
 
     def differentiate(self):
-        payload_location = self.clone_differentiation_payload()
+        payload_location = self.clone_payload()
         self.ansible.run_playbook(payload_location + self.PAYLOAD_ENTRY_ANSIBLE_PLAYBOOK)
 
-    def clone_differentiation_payload(self):
+    def clone_payload(self):
+        private_key = self.user_data_gateway.get_version_control_private_key()
+        if private_key:
```

Comment:
In general I find an `if/else` provides an easier to read structure in this context than an early return, but I'm probably just imparting a personal preference now